pilot/pkg/xds/endpoint_builder.go

```diff
@@ -132,6 +140,7 @@ func (b EndpointBuilder) Key() string {
 		strconv.FormatBool(b.clusterLocal),
 		util.LocalityToString(b.locality),
 		b.tunnelType.ToString(),
+		b.nodeName,
```

Comment:
Can be added only when internal traffic policy is enabled and service has NodeLocal policy set

Reply:
> Can be added only when internal traffic policy is enabled and service has NodeLocal policy set

yes, done